lib/portlayer/exec/container.go

```diff
@@ -475,12 +476,22 @@ func (c *Container) stop(ctx context.Context, waitTime *int32) error {
 				}
 				log.Warnf("invalid power state during power off: %s", terr.ExistingState)
 
+			case *types.GenericVmConfigFault:
+
+				// Check if the poweroff task was canceled due to a concurrent guest shutdown
+				if len(terr.FaultMessage) > 0 && terr.FaultMessage[0].Key == vmNotSuspendedKey {
+					log.Infof("power off %s task skipped due to guest shutdown", c.ExecConfig.ID)
+					return nil
+				}
+				log.Warnf("hard power off failed due to: %#v", terr)
```

Comment:
Follow-up: this doesn't log the specific typed error, which is useful for debugging intermittent failures such as the one in https://ci.vmware.run/vmware/vic/5831. I've posted a change in #2732.